File: C#/28. Find the Index of the First Occurrence in a String.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        int m=haystack.Length;
        int n=needle.Length;
        for(int i=0;i<m-n+1;i++){
            int j=0;
            while(j<n && haystack[i+j]==needle[j]){
                j++;
            }
            if(j==n){
                return i;
            }
        }
        return -1;
    }
}

File: Java/28. Find the Index of the First Occurrence in a String.java
class Solution {
    public int strStr(String haystack, String needle) {
        int m=haystack.length();
        int n=needle.length();
        for(int i=0;i<m-n+1;i++){
            int j=0;
            while(j<n && haystack.charAt(i+j)==needle.charAt(j)){
                j++;
            }
            if(j==n){
                return i;
            }
        }
        return -1;
    }
}
